buildSrc/src/main/java/org/elasticsearch/gradle/internal/InternalDistributionBwcSetupPlugin.java

```diff
@@ -279,7 +279,9 @@ static void createBuildBwcTask(
                         + "."
                         + extension
                 ),
-                expandedDistDirSupport ? new File(checkoutDir, baseDir + "/" + name + "/build/install/") : null
+                expandedDistDirSupport
+                    ? new File(checkoutDir, baseDir + "/" + name + "/build/install/elasticsearch-" + version + "-SNAPSHOT")
```

Comment:
We need to include the distribution type in the path as well (i.e. `-oss` for the OSS distro). That's what's causing these errors:

```Caused by: java.nio.file.NoSuchFileException: /dev/shm/elastic+elasticsearch+pull-request-bwc/distribution/bwc/minor/build/bwc/checkout-7.x/distribution/archives/oss-linux-tar/build/install/elasticsearch-7.12.0-SNAPSHOT/plugins/config```

Reply:
And posibly architecture too? Essentially, we need to use the same logic for building that dir name as we do for the file name as I "think" we add all those qualifiers there as well.

Reply:
I fixed the issue with the file not found above. The problem was not with the folder name but with the right declaration of the target install root which is still "build/install". The distro base folder name doesn't contain oss or anything there I think.